Add tutorial "Building OpenSSH to work with U2F Security Keys on Debian 10 (Buster)" (#251)
hetzneronline/community-content: tutorials/openssh-with-u2f-security-key-debian-10/01.en.md

FIDO2 and U2F (Universal 2 Factor) are two protocols that form the basis behind how your key authenticates with the web and other internet protocols.

In this tutorial we will be focusing on the support that OpenSSH has for these protocols, and getting it set up for you to authenticate to your Hetzner server or cloud instance running Debian 10 (Stable).

To do this we'll need to **compile** and **build** the latest version (8.3 at time of writing) of OpenSSH from the next version of Debian (11/Bullseye) in a way that it will work nicely with Debian 10. It should behave fine since it is just a simple backport.

Since this tutorial gets involved in the build process of Debian packages, it may seem a little complicated or involved at times. That said, these instructions are designed to be used by people of any skill level, and it's likely you'll learn something new along the way! :)

**Prerequisites**

To follow this tutorial you will need:

> - A Hetzner server running Debian 10, either dedicated or from Hetzner Cloud.

Reviewer: It works on any server, no matter which hosting provider ;)

Author: Good point hahah, I've edited it to show this 😛

- A local Linux installation or Windows installation (With Ubuntu or Debian installed through Windows Subsystem for Linux) 
    - Arch Linux, Debian 10+ and Ubuntu 20.04+ are known to be compatible, anything else may work if it provides a version of OpenSSH 8.2 or later. macOS is untested.
- A U2F compatible hardware security key, for example a YubiKey or Solo Key.

> While writing this tutorial it was tested on a fresh Hetzner Cloud CPX11 instance however it will run perfectly fine on other cloud instances or dedicated servers.

Reviewer: ```suggestion
While writing this tutorial it was tested on a fresh Hetzner Cloud CPX11 instance however it will run perfectly fine on any other server.
```

Author: Done, thanks!

If there are any prerequisites for your tutorial, please write them out here.

> If there is already a tutorial explaining one of the prerequisites, make sure to link to that other tutorial.

Reviewer: Please remove these lines from the template

Author: Should be done now :)
